model/provisioning.go

```diff
@@ -0,0 +1,368 @@
+package model
+
+// Section: provisioning
+
+import (
+	"fmt"
+	"github.com/gosimple/slug"
+	"net/url"
+	"time"
+)
+
+const (
+	GOS_TYPE_POSIX                 GroupOnSystemType = "POSIX_GROUP"
+	GOS_TYPE_GROUP_OF_NAMES        GroupOnSystemType = "GROUP_OF_NAMES"
+	GOS_TYPE_GROUP_OF_UNIQUE_NAMES GroupOnSystemType = "GROUP_OF_UNIQUE_NAMES"
+	GOS_TYPE_GROUP                 GroupOnSystemType = "GROUP"
+	GOS_TYPE_AZURE_ROLE            GroupOnSystemType = "AZURE_ROLE"
+	GOS_TYPE_AZURE_UNIFIED_GROUP   GroupOnSystemType = "AZURE_UNIFIED_GROUP"
+	GOS_TYPE_AZURE_SECURITY_GROUP  GroupOnSystemType = "AZURE_SECURITY_GROUP"
+
+	PSYSTEM_TYPE_LDAP                             ProvisionedSystemType = "LDAP"
+	PSYSTEM_TYPE_ACTIVE_DIRECTORY                 ProvisionedSystemType = "ACTIVE_DIRECTORY"
+	PSYSTEM_TYPE_AZURE_TENANT                     ProvisionedSystemType = "AZURE_TENANT"
+	PSYSTEM_TYPE_SOURCE_LDAP_DIRECTORY            ProvisionedSystemType = "SOURCE_LDAP_DIRECTORY"
+	PSYSTEM_TYPE_SOURCE_AZURE_OIDC_DIRECTORY      ProvisionedSystemType = "SOURCE_AZURE_OIDC_DIRECTORY"
+	PSYSTEM_TYPE_SOURCE_AZURE_SYNC_LDAP_DIRECTORY ProvisionedSystemType = "SOURCE_AZURE_SYNC_LDAP_DIRECTORY"
+	PSYSTEM_TYPE_INTERNAL_LDAP                    ProvisionedSystemType = "INTERNAL_LDAP"
+)
+
+// GroupOnSystemType Use constants as enum for type
+type GroupOnSystemType string
+
+// ProvisionedSystemType Use constants as enum for type
+type ProvisionedSystemType string
+
+// NewGroupOnSystemList Initialize a new GroupOnSystemList
+func NewGroupOnSystemList() *GroupOnSystemList {
+	return &GroupOnSystemList{
+		DType: "LinkableWrapper",
+	}
+}
+
+// GroupOnSystemList List of GroupOnSystems
+type GroupOnSystemList struct {
+	DType string          `json:"$type,omitempty"`
```

Comment:
Is this property really necessary? On the server the LinkableWrapper doesn't have this property.

Reply:
Die lijkt idd overbodig en is nu weg